libr/anal/dwarf_process.c

```diff
@@ -1071,6 +1080,9 @@ static VariableLocation *parse_dwarf_location (Context *ctx, const RBinDwarfAttr
 			reg_name = get_dwarf_reg_name (ctx->anal->cpu, reg_num, &kind, ctx->anal->bits);
 		} break;
 		case DW_OP_addr: {
+			if (i == block.length - 1) {
```

Comment:
Why is it unnecessary? Are you depending on the `if (block.length - i < addr_size)` check?

Reply:
But I think maybe the check is off by one?

Reply:
> because there is no leb128() function

Emm the `r_read_ble##()` functions, they don't do bounds-checking (as can be seen from their argument list).

> But I think maybe the check is off by one?

Suppose `i == block.length - 1` and the check is
https://github.com/radareorg/radare2/blob/1cae744fd5487d1b06b37b4bbbdc7e6273e353c6/libr/anal/dwarf_process.c#L1078-L1081

Substituting `block.length - 1` for `i`:

```c
if (block.length - (block.length - 1) < addr_size) {
        return NULL;
}
```

Simplifying:

```c
if (1 < addr_size) {
        return NULL;
}
```

The `return NULL` must happen, otherwise there's going to be an OOB read. Now, `addr_size` is defined as:

https://github.com/radareorg/radare2/blob/1cae744fd5487d1b06b37b4bbbdc7e6273e353c6/libr/anal/dwarf_process.c#L1076

So the question is whether `addr_size` can be 0 i.e. `ctx->anal->bits` is less than 8.

Now I think this can happen, whether by accident or malice, so I'd like to keep the `if (i == block.length - 1)` check (line 1086 above). Do you agree?

Reply:
> So the question is whether `addr_size` can be 0 i.e. `ctx->anal->bits` is less than 8.

This should be:

So the question is whether `addr_size` can be 0 or 1 i.e. `ctx->anal->bits` is less than 16.

But the above makes the case for keeping the check even stronger imo.

Reply:
Ah, silly me

> I don't think it's necessary here

Okay 8cfe33c28c708076e626715c94e04e92a5107f94

Reply:
Sure, makes sense. I agree